go-controller/pkg/ovn/egressip_test.go

```diff
@@ -3714,6 +3755,227 @@ var _ = ginkgo.Describe("OVN master EgressIP Operations", func() {
+
+				_, err = fakeOvn.fakeClient.KubeClient.CoreV1().Nodes().Create(context.TODO(), &node2, metav1.CreateOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Eventually(getEgressIPAllocatorSizeSafely).Should(gomega.Equal(2))
+				gomega.Eventually(isEgressAssignableNode(node1.Name)).Should(gomega.BeTrue())
+				gomega.Eventually(isEgressAssignableNode(node2.Name)).Should(gomega.BeTrue())
+				gomega.Expect(fakeOvn.controller.eIPC.allocator.cache).To(gomega.HaveKey(node1.Name))
+				gomega.Expect(fakeOvn.controller.eIPC.allocator.cache).To(gomega.HaveKey(node2.Name))
+
+				cachedEgressNode1 := fakeOvn.controller.eIPC.allocator.cache[node1.Name]
+				cachedEgressNode2 := fakeOvn.controller.eIPC.allocator.cache[node2.Name]
+				gomega.Expect(cachedEgressNode1.egressIPConfig.V6.Net).To(gomega.Equal(ip1V6Sub))
+				gomega.Expect(cachedEgressNode2.egressIPConfig.V4.Net).To(gomega.Equal(ip2V4Sub))
+
+				// Explicitly call check reachibility so we need not to wait for slow periodic timer
+				checkEgressNodesReachabilityIterate(fakeOvn.controller)
+				gomega.Expect(cachedEgressNode1.isReachable).To(gomega.BeTrue())
+				gomega.Expect(cachedEgressNode2.isReachable).To(gomega.BeTrue())
+
+				tests := []struct {
+					desc            string
+					node1FailProbes bool
+					node2FailProbes bool
+					init            func(hcc1, hcc2 *fakeEgressIPHealthClient)
+				}{
+					{
+						desc:            "disconnect nodes",
+						node1FailProbes: true,
+						node2FailProbes: true,
+						init: func(hcc1, hcc2 *fakeEgressIPHealthClient) {
+							hcc1.Disconnect()
+							hcc2.Disconnect()
+						},
+					},
+					{
+						desc:            "connect node1",
+						node2FailProbes: true,
+					},
+					{
+						desc: "node1 connected, connect node2",
+					},
+					{
+						desc: "node1 and node2 connected",
+					},
+					{
+						desc:            "node1 and node2 connected, bump only node2 counters",
+						node1FailProbes: true,
+					},
+					{
+						desc:            "node2 connected, disconnect node1",
+						node1FailProbes: true,
+						node2FailProbes: true,
+						init: func(hcc1, hcc2 *fakeEgressIPHealthClient) {
+							hcc1.Disconnect()
+						},
+					},
+					{
+						desc:            "connect node1, disconnect node2",
+						node2FailProbes: true,
+						init: func(hcc1, hcc2 *fakeEgressIPHealthClient) {
+							hcc2.Disconnect()
+						},
+					},
+					{
+						desc: "node1 and node2 connected and both counters bump",
+						init: func(hcc1, hcc2 *fakeEgressIPHealthClient) {
+							// Perform an additional iteration, to make probe counters to bump on second call
+							checkEgressNodesReachabilityIterate(fakeOvn.controller)
+						},
+					},
+				}
+				hcc1 := cachedEgressNode1.healthClient.(*fakeEgressIPHealthClient)
+				hcc2 := cachedEgressNode2.healthClient.(*fakeEgressIPHealthClient)
+				ttIterCheck := func(hcc *fakeEgressIPHealthClient, prevNodeIsConnected bool, prevProbes int, failProbes bool, desc string) {
```

Comment:
maybe some comments here explaining what this function is for

Reply:
ack. done.